packages/components/src/Badge/Badge.component.js

```diff
@@ -19,32 +21,87 @@ function Badge({
 	t,
 	style,
 	className,
+	display = 'large',
+	aslink,
+	edit,
+	icon,
+	white,
 }) {
+	const displayClass = display === 'small' ? 'tc-badge-display-small' : 'tc-badge-display-large';
+
 	const containerClasses = classNames(
 		'tc-badge',
 		theme['tc-badge'],
 		selected && ['tc-badge-selected', theme['tc-badge-selected']],
 		disabled && ['tc-badge-disabled', theme['tc-badge-disabled']],
 		!onDelete && ['tc-badge-readonly', theme['tc-badge-readonly']],
+		[displayClass, theme[displayClass]],
```

Comment:
i suggest you to use the css util that i've set on skeleton component (https://github.com/Talend/ui/blob/master/packages/components/src/Skeleton/Skeleton.component.js#L10)
(https://github.com/Talend/ui/pull/2109)

That means you could use it this way :
```javascript
// top of the file
import badgeCssModule from './Badge.scss';
import { getTheme } from '../theme';

const theme = getTheme(badgeCssModule);

// here
const containerClasses = theme(
  'tc-badge',
  displayClass,
  className,
  {
    'tc-badge-selected': selected,
    'tc-badge-disabled': disabled,
    'tc-badge-readonly': !onDelete,
    'tc-badge-aslink': aslink,
    'tc-badge-edit': edit,
  }
)
```

It allow to not repeat the class twice & using the classname style api with the object to apply the classes or not

Reply:
Of course, it applies for each classename const added in the file :)

Reply:
👍